pkg/apis/projectcalico/v3/hostendpoint.go

```diff
@@ -88,6 +88,8 @@ type EndpointPort struct {
 	Name     string               `json:"name" validate:"portName"`
 	Protocol numorstring.Protocol `json:"protocol"`
 	Port     uint16               `json:"port" validate:"gt=0"`
+	HostPort uint16               `json:"hostPort"`
```

Comment:
Hm, this is an interesting one since the structure is shared between HostEndpoints and WorkloadEndpoints, and we don't really want these fields on HostEndpoints IIUC.

We either need to:
- Add API validation in libcalico-go to make sure that HostEndpoints don't get these fields set.
- Split out a new structure - e.g., `WorkloadEndpointPort` - specifically for WEPs that has the new fields.

I think I am slightly in favor of the second option, although it might require some additional downstream changes to use the new type, and might be a tad more work. WDYT?

Would like @fasaxc to chime in if he has thoughts on it.

Reply:
Also, do we need the HostIP field? Isn't that already available through the Node API?

Reply:
Agree, probably best to split the fields rather than have an ambiguous entry on the HostEndpoint (which presumably we wouldn't actually implement)

Reply:
Thanks for the review, I didn't realize this struct was also used by HostEndpoints, we'll split out a new struct.
Regarding `HostIP`, I think we need it for multi-homed hosts where the IP to bind to may not be the node IP.

Reply:
Looks like the [Pod API allows that to be specified](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#containerport-v1-core) so we probably need to pass the host IP through.

Reply:
> Looks like the Pod API allows that to be specified so we probably need to pass the host IP through.


d'oh, right. Yeah that makes sense.